src/experimental/tools/mediaCapabilitiesProber/probers/HDCPPolicy/index.ts

```diff
@@ -30,60 +30,64 @@ export type IMediaKeyStatus =
   "internal-error";
 
 /**
+ * @param {Object} config
  * @returns {Promise}
  */
-function isHDCPAPIAvailable(): Promise<void> {
-  return new Promise<void>((resolve) => {
-    if (!("requestMediaKeySystemAccess" in navigator)) {
+export default function probeHDCPPolicy(
+  config: IMediaConfiguration
+): Promise<[ProberStatus]> {
+
+  return new Promise((resolve) => {
+    if (requestMediaKeySystemAccess == null) {
       throw new Error("API_AVAILABILITY: MediaCapabilitiesProber >>> API_CALL: " +
         "API not available");
     }
-    resolve();
-  }).then(() => {
     if (!("MediaKeys" in window)) {
       throw new Error("MediaCapabilitiesProber >>> API_CALL: " +
         "MediaKeys API not available");
     }
-    if (!("getStatusForPolicy" in (window as any).MediaKeys as any)) {
+    if (!("getStatusForPolicy" in ((window as any).MediaKeys as any).prototype)) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4325))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=365&issues=AWbvUu_fsSodgw_KZeng&open=AWbvUu_fsSodgw_KZeng)

[ISSUE_KEY]: AWbvUu_fsSodgw_KZeng

Reply:
I think it would be better to use MediaKeys_ here